flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateOrAlterMaterializedTable.java

```diff
@@ -0,0 +1,160 @@
+                refreshMode,
+                asQuery);
+        this.isOrAlter = isOrAlter;
+    }
+
+    @Override
+    public SqlOperator getOperator() {
+        return isOrAlter ? CREATE_OR_ALTER_OPERATOR : CREATE_OPERATOR;
+    }
+
+    public boolean isOrAlter() {
+        return isOrAlter;
+    }
+
+    @Override
+    public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+        writer.keyword("CREATE");
+        if (isOrAlter) {
+            writer.keyword("OR ALTER");
+        }
+        writer.keyword("MATERIALIZED TABLE");
+        getTableName().unparse(writer, leftPrec, rightPrec);
+
+        if (!getColumnList().isEmpty()
+                || !getTableConstraints().isEmpty()
+                || getWatermark().isPresent()) {
+            SqlUnparseUtils.unparseTableSchema(
+                    writer,
+                    leftPrec,
+                    rightPrec,
+                    getColumnList(),
+                    getTableConstraints(),
+                    getWatermark().orElse(null));
+        }
+
+        getComment()
+                .ifPresent(
+                        comment -> {
+                            writer.newlineAndIndent();
+                            writer.keyword("COMMENT");
+                            comment.unparse(writer, leftPrec, rightPrec);
+                        });
+
+        if (getDistribution() != null) {
+            writer.newlineAndIndent();
+            getDistribution().unparse(writer, leftPrec, rightPrec);
+        }
+
+        if (!getPartitionKeyList().isEmpty()) {
+            writer.newlineAndIndent();
+            writer.keyword("PARTITIONED BY");
+            SqlWriter.Frame partitionedByFrame = writer.startList("(", ")");
+            getPartitionKeyList().unparse(writer, leftPrec, rightPrec);
+            writer.endList(partitionedByFrame);
+        }
+
+        if (!getPropertyList().isEmpty()) {
+            writer.newlineAndIndent();
+            writer.keyword("WITH");
+            SqlWriter.Frame withFrame = writer.startList("(", ")");
+            for (SqlNode property : getPropertyList()) {
+                SqlUnparseUtils.printIndent(writer);
+                property.unparse(writer, leftPrec, rightPrec);
+            }
+            writer.newlineAndIndent();
+            writer.endList(withFrame);
+        }
+
+        if (getFreshness() != null) {
+            writer.newlineAndIndent();
+            writer.keyword("FRESHNESS");
+            writer.keyword("=");
+            getFreshness().unparse(writer, leftPrec, rightPrec);
+        }
+
+        if (getRefreshMode() != null) {
+            writer.newlineAndIndent();
+            writer.keyword("REFRESH_MODE");
+            writer.keyword("=");
+            writer.keyword(getRefreshMode().name());
+        }
+
+        writer.newlineAndIndent();
+        writer.keyword("AS");
+        writer.newlineAndIndent();
+        getAsQuery().unparse(writer, leftPrec, rightPrec);
+    }
```

Comment:
I guess it is duplicating what we have in `SqlCreateMaterializedTable`
I wonder if we can extract this part into a separate method and reuse for both?